CorePayments/src/main/java/com/paypal/android/corepayments/analytics/AnalyticsEventData.kt

```diff
@@ -5,5 +5,6 @@ internal data class AnalyticsEventData(
     val environment: String,
     val eventName: String,
     val timestamp: Long,
-    val orderId: String?
+    val orderId: String?,
+    val buttonType: String? = null
```

Comment:
q: Can we put the button type into the event name? That way we wouldn't need an extra property.

Reply:
It's a local data holder that's eventually converted to event data https://github.com/paypal/paypal-android/pull/245/files/07dd656c767aed4d03e04c384ada2d099fb45eb0#diff-65c7e120a418bd43366952c069e36e07ede55b6ca30b1bfebfecbdafe3443db4R49.

I'm not sure there's an advantage to appending it to the event name as we'll have to parse it out and send it separately in the event data anyway. Please let me know if I'm missing something.

Reply:
Got it ok. FPTI may have more features than I realize. For BT we would flatten everything to a string.

Reply:
I think generally our approach with FPTI has been to keep the event names streamlined and add any extra data as a tag. This makes it easier to calculate conversion of a feature as a whole (counting all button conversion as a whole) as well as drill into the data (via tags) for the difference in conversion of things like paypal vs pay later vs credit. This prevents us from needing to do more math by adding slightly different event names together to calculate a whole feature.

Reply:
@jaxdesmarais to be clear, you want the `button_type` to be a param/tag and not concatenate `button_type` with the event name, correct?

Reply:
Yep, how you have it, was just explaining a bit more of the reasoning to @sshropshire of why we have moved to utilizing tags vs throwing everything in the event name like we did for previous version of BT since I had the context. 😄